### PR TITLE
Fixed

### LINKED ISSUE #372: Looting of Crazed Summer Rearguards
**Describe the bug**
When trying to open a dead crazed summer rearguard, the client shuts down and logs the player out.

**To Reproduce**
Go to the court of summer.
Kill a crazed rearguard.
Open the corpse. 

**Expected behavior**
The body should open and display the loot. 

Screenshot (3)

### PR DESCRIPTION
**Pull request by ghost**
- added elf corpse by oualid6496
- moved food storages to lib file
- removed duplicated function from lib file
- updated config lua to client 12 instead of 11.49
- fixed a few item typos

Co-Authored-By: oualid6496 <32771228+oualid6496@users.noreply.github.com>

### CODE REVIEW

File: data/movements/scripts/others/drowning.lua [hunk 1]
```diff
@@ -1,43 +1,45 @@
...
-         player:setStorageValue(17101,0)
-       end
-     end
-   end
-   return true
-end
\ No newline at end of file
+function onStepOut(creature, item, position, fromPosition)
+	local player = creature:getPlayer()
+	if player == nil then
```
Comment (gpedro) [inline data/movements/scripts/others/drowning.lua]: change to `if not player then`

File: data/movements/scripts/others/drowning.lua [hunk 2]
```diff
@@ -1,43 +1,45 @@
...
-     if player:getStorageValue(17100) > os.time() then
-     if player:getStorageValue(17101) < 1 then
-       doChangeSpeed(cid, 600)
-       player:setStorageValue(17101,1)
-     end
-     end
-   end
+function onStepIn(creature, item, position, fromPosition)
+	local player = creature:getPlayer()
+	if player == nil then
```
Comment (gpedro) [inline data/movements/scripts/others/drowning.lua]: change to `if not player then`

File: data/movements/scripts/others/drowning.lua [hunk 3]
```diff
@@ -1,43 +1,45 @@
...
+			player:changeSpeed(600)
+			player:setStorageValue(Storage.DrowningSpeed,1)
+		end
+	end
 
-   if(math.random(1, 10) == 1) then
-     Position(pos):sendMagicEffect(CONST_ME_BUBBLES)
-   end
-   return true
+	if(math.random(1, 10) == 1) then
```
Comment (gpedro) [inline data/movements/scripts/others/drowning.lua]: `math.random(1,10)` can be replaced to `math.random(10)`

File: data/movements/scripts/others/drowning.lua [hunk 4]
```diff
@@ -1,43 +1,45 @@
...
+function onStepIn(creature, item, position, fromPosition)
+	local player = creature:getPlayer()
+	if player == nil then
+		return true
+	end
+	if player:getCondition(CONDITION_DROWN,CONDITIONID_COMBAT) == nil then
+		player:addCondition(condition)
+	end
+	if player:getStorageValue(Storage.DrowningSpeed) > os.time() then
+		if player:getStorageValue(Storage.DrowningSpeed) < 1 then
```
Comment (gpedro) [inline data/movements/scripts/others/drowning.lua]: this condition can be merge as once.

File: data/movements/scripts/others/drowning.lua [hunk 5]
```diff
@@ -1,43 +1,45 @@
...
\ No newline at end of file
+function onStepOut(creature, item, position, fromPosition)
+	local player = creature:getPlayer()
+	if player == nil then
+		return true
+	end
+	local ppos = player:getPosition()
+	if(isInArray(underWater, getThingfromPos({x = ppos.x, y = ppos.y, z = ppos.z, stackpos = 0}).itemid)) then
+		return true
+	else
```
Comment (gpedro) [inline data/movements/scripts/others/drowning.lua]: this else-end is unuseless.

File: data/actions/scripts/others/consumables/cupcakes.lua [hunk 6]
```diff
@@ -1,32 +1,31 @@
...
-    else
-         doPlayerSendCancel(cid, "You are full or you've already eaten a dish within 10 minutes.")
-    end
-    return true
-end
\ No newline at end of file
+function onUse(player, item, fromPosition, target, toPosition, isHotkey)
+	if (player:getStorageValue(Storage.CupCakes) <= os.time()) then
+		if (item.itemid == 31719) then
+			Creature(cid):addMana(Creature(cid):getMaxMana() - Creature(cid):getMana())
```
Comment (gpedro) [inline data/actions/scripts/others/consumables/cupcakes.lua]: what's your think about to store `Creature(cid)` on some variable?

File: data/movements/scripts/others/drowning.lua (hunk 3 above)
Comment (ghost) [reply data/movements/scripts/others/drowning.lua]: (10) não é do 0 ao 10?

File: data/movements/scripts/others/drowning.lua (hunk 4 above)
Comment (ghost) [reply data/movements/scripts/others/drowning.lua]: como?

File: data/movements/scripts/others/drowning.lua (hunk 3 above)
Comment (gpedro) [reply data/movements/scripts/others/drowning.lua]: pior que não...

Captura de Tela 2019-08-17 às 19 54 21


documentação: http://lua-users.org/wiki/MathLibraryTutorial

File: data/movements/scripts/others/drowning.lua (hunk 4 above)
Comment (gpedro) [reply data/movements/scripts/others/drowning.lua]:
```
local drowningSpeed = player:getStorageValue(Storage.DrowningSpeed)
if drowningSpeed > os.time() and drowningSpeed < 1 then
```

File: data/actions/scripts/others/consumables/cupcakes.lua (hunk 6 above)
Comment (EvilHero90) [reply data/actions/scripts/others/consumables/cupcakes.lua]: `player:addMana(player:getMaxMana() - player:getMana())`

File: data/actions/scripts/others/consumables/cupcakes.lua [hunk 7]
```diff
@@ -1,32 +1,31 @@
...
+			player:say("Mmmm.",TALKTYPE_MONSTER_SAY)
+			item:remove(1)
+			player:setStorageValue(Storage.CupCakes, os.time() + config.exhaust)
+		elseif (item.itemid == 31720) then
+			Creature(cid):addHealth(getCreatureMaxHealth(cid) - getCreatureHealth(cid))
```
Comment (EvilHero90) [inline data/actions/scripts/others/consumables/cupcakes.lua]: `player:addHealth(player:getMaxHealth() - player:getHealth())`

File: data/actions/scripts/others/consumables/cupcakes.lua [hunk 8]
```diff
@@ -1,32 +1,31 @@
...
+			player:say("Mmmm.",TALKTYPE_MONSTER_SAY)
+			item:remove(1)
+			player:setStorageValue(Storage.CupCakes, os.time() + config.exhaust)
+		elseif (item.itemid == 31721) then
+			player:say("Mmmm.",TALKTYPE_MONSTER_SAY)
+			cid:addCondition(lemon)
```
Comment (EvilHero90) [inline data/actions/scripts/others/consumables/cupcakes.lua]: `player:addCondition(lemon)`

File: data/actions/scripts/others/consumables/hirelingfood.lua [hunk 9]
```diff
@@ -27,53 +26,53 @@ local carrotPie = Condition(CONDITION_ATTRIBUTES)
...
 carrotPie:setParameter(CONDITION_PARAM_SKILL_DISTANCE, 7)
 
-function onUse(player, item, frompos, item2, topos)
+function onUse(player, item, fromPosition, target, toPosition, isHotkey)
 	local cid = player:getId()
-	if (player:getStorageValue(config.storage) <= os.time()) then
+	if (player:getStorageValue(Storage.HirelingFood) <= os.time()) then
 		if (item.itemid == 35172) then
 			player:say("Chomp.",TALKTYPE_MONSTER_SAY)
 			Creature(cid):addCondition(roastedWyvernWings)
```
Comment (EvilHero90) [inline data/actions/scripts/others/consumables/hirelingfood.lua]: `player:addCondition`

File: data/actions/scripts/others/consumables/hirelingfood.lua [hunk 10]
```diff
@@ -27,53 +26,53 @@ local carrotPie = Condition(CONDITION_ATTRIBUTES)
...
 			item:remove(1)
-			player:setStorageValue(config.storage, os.time() + config.exhaust)
+			player:setStorageValue(Storage.HirelingFood, os.time() + config.exhaust)
 		elseif (item.itemid == 35173) then
 			player:say("Yummm.",TALKTYPE_MONSTER_SAY)
 			Creature(cid):addCondition(carrotPie)
```
Comment (EvilHero90) [inline data/actions/scripts/others/consumables/hirelingfood.lua]: `player:addCondition`

File: data/actions/scripts/others/consumables/hirelingfood.lua [hunk 11]
```diff
@@ -27,53 +26,53 @@ local carrotPie = Condition(CONDITION_ATTRIBUTES)
...
 			item:remove(1)
-			player:setStorageValue(config.storage, os.time() + config.exhaust)
+			player:setStorageValue(Storage.HirelingFood, os.time() + config.exhaust)
 		elseif (item.itemid == 35174) then
 			player:say("Munch.",TALKTYPE_MONSTER_SAY)
 			Creature(cid):addCondition(tropicalMarinatedTiger)
```
Comment (EvilHero90) [inline data/actions/scripts/others/consumables/hirelingfood.lua]: `player:addCondition`

File: data/actions/scripts/others/consumables/hirelingfood.lua [hunk 12]
```diff
@@ -27,53 +26,53 @@ local carrotPie = Condition(CONDITION_ATTRIBUTES)
...
 			item:remove(1)
-			player:setStorageValue(config.storage, os.time() + config.exhaust)
+			player:setStorageValue(Storage.HirelingFood, os.time() + config.exhaust)
 		elseif (item.itemid == 35175) then
 			player:say("Munch.",TALKTYPE_MONSTER_SAY)
 			Creature(cid):addCondition(delicatessenSalad)
```
Comment (EvilHero90) [inline data/actions/scripts/others/consumables/hirelingfood.lua]: `player:addCondition`

File: data/actions/scripts/others/consumables/hirelingfood.lua [hunk 13]
```diff
@@ -27,53 +26,53 @@ local carrotPie = Condition(CONDITION_ATTRIBUTES)
...
 			item:remove(1)
-			player:setStorageValue(config.storage, os.time() + config.exhaust)
+			player:setStorageValue(Storage.HirelingFood, os.time() + config.exhaust)
 		elseif (item.itemid == 35176) then
 			player:say("Yummm.",TALKTYPE_MONSTER_SAY)
 			Creature(cid):addCondition(chilliConCarniphila)
```
Comment (EvilHero90) [inline data/actions/scripts/others/consumables/hirelingfood.lua]: `player:addCondition`

File: data/actions/scripts/others/consumables/hirelingfood.lua [hunk 14]
```diff
@@ -27,53 +26,53 @@ local carrotPie = Condition(CONDITION_ATTRIBUTES)
...
 			item:remove(1)
-			player:setStorageValue(config.storage, os.time() + config.exhaust)
+			player:setStorageValue(Storage.HirelingFood, os.time() + config.exhaust)
 		elseif (item.itemid == 35177) then
 			player:say("Mmmmm.",TALKTYPE_MONSTER_SAY)
 			Creature(cid):addCondition(svargrondSalmonFilet)
```
Comment (EvilHero90) [inline data/actions/scripts/others/consumables/hirelingfood.lua]: `player:addCondition`

File: data/actions/scripts/others/consumables/hirelingfood.lua [hunk 15]
```diff
@@ -27,53 +26,53 @@ local carrotPie = Condition(CONDITION_ATTRIBUTES)
...
 			item:remove(1)
-			player:setStorageValue(config.storage, os.time() + config.exhaust)
+			player:setStorageValue(Storage.HirelingFood, os.time() + config.exhaust)
 		elseif (item.itemid == 35178) then
 			Creature(cid):addHealth(getCreatureMaxHealth(cid)  * 0.3)
```
Comment (EvilHero90) [inline data/actions/scripts/others/consumables/hirelingfood.lua]: `player:addHealth(player:getMaxHealth() * 0.3)`

File: data/actions/scripts/others/consumables/hirelingfood.lua [hunk 16]
```diff
@@ -27,53 +26,53 @@ local carrotPie = Condition(CONDITION_ATTRIBUTES)
...
 			player:say("Munch.",TALKTYPE_MONSTER_SAY)
 			item:remove(1)
-			player:setStorageValue(config.storage, os.time() + config.exhaust)
+			player:setStorageValue(Storage.HirelingFood, os.time() + config.exhaust)
 		elseif (item.itemid == 35179) then
 			Creature(cid):addMana(Creature(cid):getMaxMana() * 0.3)
```
Comment (EvilHero90) [inline data/actions/scripts/others/consumables/hirelingfood.lua]: `player:addMana(player:getMaxMana() * 0.3)`

**Review by EvilHero90 — CHANGES_REQUESTED** [review]
all of them can be converted

File: data/actions/scripts/others/consumables/cupcakes.lua (hunk 6 above)
Comment (ghost) [reply data/actions/scripts/others/consumables/cupcakes.lua]: oh yeah I really forgot about it

File: data/movements/scripts/others/drowning.lua (hunk 5 above)
Comment (ghost) [reply data/movements/scripts/others/drowning.lua]: ve se ta certo essa parte agora

File: data/actions/scripts/others/consumables/cupcakes.lua [hunk 17]
```diff
@@ -1,32 +1,34 @@
...
-end
\ No newline at end of file
+function onUse(player, item, fromPosition, target, toPosition, isHotkey)
+	if (player:getStorageValue(Storage.CupCakes) <= os.time()) then
+		if (item.itemid == 31719) then
+			player:addMana(player:getMaxMana() - player:getMana())
+			player:say("Mmmm.",TALKTYPE_MONSTER_SAY)
+			fromPosition:sendMagicEffect(CONST_ME_MAGIC_RED)
+			item:remove(1)
+			player:setStorageValue(Storage.CupCakes, os.time() + config.exhaust)
```
Comment (gpedro) [inline data/actions/scripts/others/consumables/cupcakes.lua]: we have thrice repeated parts. we cannot reuse? repeated part:
```lua
			player:say("Mmmm.",TALKTYPE_MONSTER_SAY)
			fromPosition:sendMagicEffect(CONST_ME_MAGIC_RED)
			item:remove(1)
			player:setStorageValue(Storage.CupCakes, os.time() + config.exhaust)
```

Comment (gpedro) [reply data/actions/scripts/others/consumables/cupcakes.lua]: something like that.

```lua
local config = {
	exhaust = 600, --Exhaust is in seconds 600 equals 10min
	allowedItems = {
		31719, -- blueberry cupcake
		31720, -- strawberry cupcake
		31721, -- lemon cupcake
	}
}

local lemon = Condition(CONDITION_ATTRIBUTES)
lemon:setParameter(CONDITION_PARAM_TICKS, 60 * 60 * 1000)
lemon:setParameter(CONDITION_PARAM_SKILL_DISTANCE, 10)

function onUse(player, item, fromPosition, target, toPosition, isHotkey)
	if player:getStorageValue(Storage.CupCakes) > os.time() then
		player:sendCancelMessage("You are full or you've already eaten a dish within 10 minutes.")
		return true
	end
	
	if not config.allowedItems[item.itemid] then
		return true
	end
	
	if (item.itemid == 31719) then
		player:addMana(player:getMaxMana() - player:getMana())
	end
	
	if (item.itemid == 31720) then
		player:addHealth(player:getMaxHealth() - player:getHealth())
	end
		
	if (item.itemid == 31721) then
		player:addCondition(lemon)
	end	

	player:say("Mmmm.",TALKTYPE_MONSTER_SAY)
	fromPosition:sendMagicEffect(CONST_ME_MAGIC_RED)
	item:remove(1)
	player:setStorageValue(Storage.CupCakes, os.time() + config.exhaust)

	return true
end

```

Comment (gpedro) [reply data/actions/scripts/others/consumables/cupcakes.lua]: ps: is not tested

File: data/actions/scripts/others/consumables/hirelingfood.lua [hunk 18]
```diff
@@ -1,79 +1,86 @@
...
+			fromPosition:sendMagicEffect(CONST_ME_MAGIC_RED)
 			item:remove(1)
-			player:setStorageValue(config.storage, os.time() + config.exhaust)
+			player:setStorageValue(Storage.HirelingFood, os.time() + config.exhaust)
 		elseif (item.itemid == 35180) then
 			player:say("Blurg.",TALKTYPE_MONSTER_SAY)
+			fromPosition:sendMagicEffect(CONST_ME_MAGIC_RED)
 			item:remove(1)
-			player:setStorageValue(config.storage, os.time() + config.exhaust)
+			player:setStorageValue(Storage.HirelingFood, os.time() + config.exhaust)
```
Comment (gpedro) [inline data/actions/scripts/others/consumables/hirelingfood.lua:83]: we can refactor using same cupcake logic... what are you think?

Comment (ghost) [reply data/actions/scripts/others/consumables/hirelingfood.lua:83]: depois, q os hirelings forem implementados :P essa PR me deu muita dor de cabeça já kkkkk quero seguir a vida em frente

File: data/lib/miscellaneous/050-functions.lua [hunk 19]
```diff
@@ -1,18 +1,3 @@
-function getAccountNumberByPlayerName(name)
```
Comment (gpedro) [inline data/lib/miscellaneous/050-functions.lua:1]: pq removeu?

Comment (ghost) [reply data/lib/miscellaneous/050-functions.lua:1]: já existe ela no compat.lua

https://github.com/opentibiabr/OTServBR-Global/blob/develop/data/lib/compat/compat.lua#L518

**Comment by vankk** [comment]
At `jeanpierrefood.lua` there is a few changes that I want to point out:
- Variable names, should fix them. There is a variable called `r` and called `r_t`
- Alot of return true on item list sequence. It is not needed.
- Instead of list sequence it could be pointed to a table for the Player.say, Player.sendTextMessage and Effect to display.

File: data/actions/scripts/others/consumables/jeanpierrefood.lua [hunk 20]
```diff
@@ -4,132 +4,137 @@ local conditions = {
...
+local condition_light = Condition(CONDITION_LIGHT)
+	condition_light:setParameter(CONDITION_PARAM_LIGHT_LEVEL, 14)
+	condition_light:setParameter(CONDITION_PARAM_LIGHT_COLOR, 154)
+	condition_light:setParameter(CONDITION_PARAM_TICKS, 60*60*1000)
+
 local iid = {[9992] = 0, [9993] = 1, [9994] = 2, [9995] = 3, [9996] = 4, [9997] = 5, [9998] = 6, [9999] = 7, [10000] = 8, [10001] = 9, [12540] = 10, [12542] = 11, [12543] = 12, [12544] = 13}
 
-function onUse(player, item, frompos, item2, topos)
+function onUse(player, item, fromPosition, target, toPosition, isHotkey)
 	local cid = player:getId()
```
Comment (vankk) [inline data/actions/scripts/others/consumables/jeanpierrefood.lua]: This line isn't been used.

File: data/creaturescripts/scripts/others/logout.lua [hunk 21]
```diff
@@ -20,6 +20,6 @@ function onLogout(player)
 		end
 	end
 	
-	player:setStorageValue(17101,0)
+	player:setStorageValue(Storage.DrowningSpeed,0)
```
Comment (vankk) [inline data/creaturescripts/scripts/others/logout.lua:23]: Add a space before the zero.

File: data/movements/scripts/others/drowning.lua [hunk 22]
```diff
@@ -1,43 +1,43 @@
...
-   return true
-end
\ No newline at end of file
+function onStepOut(creature, item, position, fromPosition)
+	local player = creature:getPlayer()
+	if not player then
+		return true
+	end
+	local ppos = player:getPosition()
+	if(isInArray(underWater, getThingfromPos({x = ppos.x, y = ppos.y, z = ppos.z, stackpos = 0}).itemid)) then
```
Comment (vankk) [inline data/movements/scripts/others/drowning.lua:34]: Why use a outdate function? You can use here `Tile.getGround`

Comment (EPuncker) [reply data/movements/scripts/others/drowning.lua]: an*

**Comment by ghost** [comment]
guys i'm no longer supporting this PR, feel free to merge it and do whatever you want to it, but i'm not touching this code again 😛

File: data/movements/scripts/others/drowning.lua (hunk 22 above)
Comment (vankk) [reply data/movements/scripts/others/drowning.lua:34]: You should be focusing on the script (alot of mistakes and outdateds functions) and not on my pronunciation.

**Comment by dudantas** [comment]
> guys i'm no longer supporting this PR, feel free to merge it and do whatever you want to it, but i'm not touching this code again 😛

Are you willing to improve the codes? if not, better not send PR.
Contribution is always good, but we're all here subject to constructive criticism and tips for improving our codes and if you don't think you can be charged for it, then I really don't think you need to submit PR.

**Comment by joseluis2g** [comment]
rofl

File: data/movements/scripts/others/drowning.lua (hunk 22 above)
Comment (ghost) [reply data/movements/scripts/others/drowning.lua:34]: a parte engraçada é que a PR do script anterior "sem meta-tables" foi aprovada sem mimimi nenhum, e agora que peguei pra fazer algo do tipo trocar doRemoveCondition por player:removeCondition, meu PR tenho q ser obrigado a otimizar o script inteiro?

Comment (ghost) [reply data/movements/scripts/others/drowning.lua:34]: e pronuncia é fala, não escrita rs

Comment (miezaF) [reply data/movements/scripts/others/drowning.lua:34]: é ruim quando é com voce né? :)

**Pull request closed without merging**